### PR TITLE
Replace Detomatix with Headset Disruptor

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

This PR removes the Detomatix Cartridge entirely from the game and replaces it with a completely new item, the Headset Disruptor.
This is an idea that I stole directly from Mothblocks on the tgstation discord.

This new item is very simple: 
You use it once in-hand and select a department (Cargo, Engineering, Medical, Science, Security, or Service).
You use it a second time and it emits an ungodly sound from every radio tuned to that department.
If that radio is being worn directly on a person then it will stun them for 6 seconds, confuse them for 15, and deafen them for 30 (unless it is turned off or they can't hear for some other reason). It also drops their radio onto the floor, as it is physically blasted off their ears or out of their pocket.
These effects are halved by wearing noise protection (such as Bowman headsets).

The item costs 6TC, is consumed upon use, and you can only buy one, so use it wisely to either get into or out of a location or situation, then get rid of the evidence.
Nuke Ops can't buy it but Clown Ops can, as it is clearly clown-adjacent.

You can think of it as a more targetted, shorter duration version of the CRAB-17 perhaps, although its utility is also somewhat different.
Disrupt people chasing you long enough to get away. Turn the tide of a riot. Blast a miner's ears just as they announce they're about to pull Bubblegum. Keep latecomers off the escape shuttle so you can hijack more easily. Send a powerful message to one guy that you think what they are saying on the radio is annoying as hell.

## Why It's Good For The Game

Detomatix:
- Very unreliable.
- Convoluted sequence of activation.
- When it works the chief damage it inflicts to anyone is destroying their ID Card which is mostly just annoying and not beneficial for the person doing it.
- Encourages people not to put their ID cards inside their PDAs, which is silly.
- Wide code influence for a feature nobody uses.

Headset Disruptor:
- Can be used offensively or defensively.
- People will definitely notice when it happens.
- Leaves forensic evidence (an object).
- Tied to an item no sane person is going to stop using just to avoid this effect.
- Significantly more collateral damage.

## Changelog

:cl: Jacquerel and Mothblocks
add: Traitors can now purchase an item which hijacks headsets to temporarily stun and deafen a specific department.
del: The Detomatix Cartridge has been removed from the game.
/:cl:

### CODE REVIEW

**Review by MrMelbert — COMMENTED** [review]
- At a glance, 8 seconds of hard stun across multiple people _seems_ busted overpowered and unfun to play against. 
   - ESPECIALLY since this is purchasable by Nuke Ops. 
   - Being able to effortlessly stun the entire security team probably lends to full security teams being very easily wiped. (Maybe bowmans should protect against it? (Or just part of it, like just the stun but not the confuse?))
- I would suggest the Captain's Headset gains immunity to this, like how the Captain's PDA had immunity to PDA bombs. To prevent Nuke Ops Cheese. (Also since they're tuned to all frequencies...) 
- Otherwise, I will find this very funny when someone accidentally blows out their own eardrums without realizing it.

File: code/game/objects/items/headset_overload.dm [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+			RADIO_CHANNEL_ENGINEERING,
+			RADIO_CHANNEL_MEDICAL,
+			RADIO_CHANNEL_SCIENCE,
+			RADIO_CHANNEL_SECURITY,
+			RADIO_CHANNEL_SERVICE,
+			RADIO_CHANNEL_SUPPLY,
+		)
+
+		var/picked = tgui_input_list(user, "Choose a channel to disrupt.", "Channel Selector", department_channels)
+		if (isnull(picked))
```
Comment (MrMelbert) [inline code/game/objects/items/headset_overload.dm:30]: Also: Input stalling
```suggestion
		if (isnull(picked) || expended) 
```
```suggestion
		if (isnull(picked) || expended || !user.is_holding(src)) // Or if you want to allow telekinesis you can use can_perform_action or whatever you want 
```

**Comment by Metekillot** [comment]
I agree with Melbert that an 8 second stun of an entire department (including Security) for 4 TC is too cheap.

**Comment by Jacquerel** [comment]
> * At a glance, 8 seconds of hard stun across multiple people _seems_ busted overpowered and unfun to play against.
>       
>       * ESPECIALLY since this is purchasable by Nuke Ops.
>       * Being able to effortlessly stun the entire security team probably lends to full security teams being very easily wiped. (Maybe bowmans should protect against it? (Or just part of it, like just the stun but not the confuse?))
> 
>     * I would suggest the Captain's Headset gains immunity to this, like how the Captain's PDA had immunity to PDA bombs. To prevent Nuke Ops Cheese. (Also since they're tuned to all frequencies...)
> 
>     * Otherwise, I will find this very funny when someone accidentally blows out their own eardrums without realizing it.

I miiight give this like a two second delay or something in which you can take off your headset, although I'm not sure if that functionally changes anything. It also probably shouldn't be available to Ops yeah.
Your other suggestions are very sensible.

**Comment by Mothblocks** [comment]
> I miiight give this like a two second delay or something in which you can take off your headset

Don't like this no

**Comment by Mothblocks** [comment]
Bowmans definitely shouldn't protect completely against it but it seems like a good idea to nerf the effect in some way, and ideally show it in the UI somehow, like "Security (Weaker)".

**Comment by Mothblocks** [comment]
And I think it's not too crazy to remove it from being able to be purchased by nuke ops either and play with that separately from this PR

**Comment by carlarctg** [comment]
Put mothblocks in the changelog credits you DISGUSTING idea thief

Also this is a cool concept and def more interesting

8 sec hardstun is clearly a bad idea - i think it should do EVERYTHING except that. it should knock them down for 3 seconds, daze them, stagger them, make them dizzy, basically render them an incoherent blubbering mess, but one that can still, in theory, fight back and maybe screech some undecipherable garbage over the radio in a plea for help

**Comment by moocowswag** [comment]
Does taking the encryption key out of your headset (losing access to department channel) prevent you from getting blasted if that department is targeted?

**Comment by Bibarra** [comment]
Why not have both? One of the fun things I like to do is pretend I'm deaf to get someone to engage in a PDA conversation with me, then blast 'em with the detomatix. Also while I like this headset disruptor thing, a wide ranging 8 second stun seems hilariously overpowered for 4tc.

**Comment by Jacquerel** [comment]
Reduced the stun a little bit and it's further reduced by wearing bowmans or other equipment which gives you sound protection.

**Comment by Jacquerel** [comment]
> Does taking the encryption key out of your headset (losing access to department channel) prevent you from getting blasted if that department is targeted?
> 
> Does it effect borgs with radio access to that department?

It only affects radios which would receive a message you send on that channel at the time it is activated.
I uh, I should check what it does for cyborgs.

**Comment by ATH1909** [comment]
this seems really really good at cheesing assassinate X person objectives (or for nicking sec gear for the necessary immunities your rampage). you find your target, hit the sec radio frequency to stun them for 3 seconds with no counterplay, warning, or ability to dodge, and rip into them with a baton/melee weapon of choice/grenade/.357 revolver while they're unable to fight back or even call for help.

if you have friends, you can conceivably straight up crit your target before they recover from the 3 second stun; if a traitor gets rev'd before they've spent all of their tc and can buy this item, that's the freest head of staff kill you've ever seen.

with the detomatix, they at least have to fall for your bait and reply to your pda message. this, meanwhile, is viable to pull out mid-combat as "fuck you I win" effect after you trade baton hits in the initial joust.

this is all assuming that your target has a bowman, of course. if they don't, they're FUCKED.

how does this interact with syndicate encryption keys?

**Comment by moocowswag** [comment]
Now that I think about it, this would be very !FUN! for rev rounds in the event a tot gets converted.

**Comment by Jacquerel** [comment]
For concerns about this being a great way to kill one guy: yeah probably it is.
Maybe I'm out of touch but I'm not _that_ worried about it, it's a one-time purchase and it's not like traitor doesn't have a number of duel-winning buttons. You can buy the contractor's baton and sure it's much more expensive but you can also use it to kill basically an unlimited number of isolated departments without alerting an entire department that someone is up to something.

**Comment by ATH1909** [comment]
> For concerns about this being a great way to kill one guy: yeah probably it is. Maybe I'm out of touch but I'm not _that_ worried about it, it's a one-time purchase and it's not like traitor doesn't have a number of duel-winning buttons. You can buy the contractor's baton and sure it's much more expensive but you can also use it to kill basically an unlimited number of isolated targets without alerting an entire department that someone is up to something.
> 
> My essential game design opinion about a one-on-one with a traitor is that it's intended that if they got the jump, they are probably going to win.

and I think the emphasis should be on the "probably". even if, realistically, your chance of winning a duel with a baton traitor with the jump on you is very low, it's still nonzero. there are still things you can reflect on/a theoretical out (just juke better 5head).

with the headset disruptor, you're utterly helpless. you can't dodge, and you have zero ability to affect the game or your fate from the moment they hotkey the disruptor out of their backpack and press the button to the moment you're critted. it is a 10-0 matchup if the other side has prepped correctly, and that FEELS bad to be on the receiving end of, even if it's technically not optimal for the traitor hunting you to buy a single use item.

**Comment by carlarctg** [comment]
the above is why I suggested the'fuck them up in every way but hardstuns' method

**Comment by zxaber** [comment]
The Detomax cart also was a basic guaranteed kill. At least this doesn't delete your ID on the off-chance that you make it out alive.

A good change all around.

**Comment by ATH1909** [comment]
> The Detomatrix cart also was a basic guaranteed kill. At least this doesn't delete your ID on the off-chance that you make it out alive.
> 
> A good change all around.

the detomatix, post-rework, still requires you to reply to a suspicious PDA message

even disregarding that, just because the detomatix is cringe doesn't mean its replacement should be

**Comment by SomeoneYouProbablyKnow** [comment]
It's not even the power level that worries me, plenty of items just kill you. It's that getting hit by something you had no way to prevent feels infinitely worse than getting hit by something you had a 1% chance to counteract. And you underestimate how afraid people are of random bullshit, many will disable their department channels and let's be honest the most commonly used channel is security, you don't want to give those lone wolf valid hunters in red even more justification to go out alone and yell help on common once things go south. Plenty of people threw away PDAs when detomax was a thing, frankly, turning off your department channel is even easier to do.

I think this should give the target some agency or at least an illusion of it. Even standard issue sleepy pens give you a few moments to do something, why can't this make you unable to use your hands and confuse you severely instead. Let them shuffle around trying to dodge or run away, they'd fail most of the time but it's better than a stun making you sit there twiddling your thumbs waiting for it to end.

Also nuckies will be able to use this item, all they need is a single traitor buying a syndi encryption key. Also will this automatically stun the whole nuckie team? They are listening to all station channels and I don't think they can turn that off easily?

**Comment by Jacquerel** [comment]
I am sympathetic to this idea that the unavoidable consequence is going to be a bad feeling moment even if it isn't necessarily out of whack with general power level of items, what I'm struggling with is how to mitigate that without essentially neutering the effect 😅

**Comment by SomeoneYouProbablyKnow** [comment]
Yea if pacificsm wasn't a horrid mess of exceptions and bypasses i'd say make it disorient and pax but you can still stun as a pacifist and that's enough to fight and win. But maybe something like a combination of not being able to use your hands, disoriente and stamina damage can do the trick? Enough debilitation to deter anything but trying to get the hell away

**Comment by IndieanaJones** [comment]
Traitors should be scary. I personally think the current version of this is fine.

**Comment by Deadgebert** [comment]
an auto win 1v1 for 6 tc?

**Comment by SomeoneYouProbablyKnow** [comment]
> Traitors should be scary. I personally think the current version of this is fine.

I mean the thig is that it's not scary. It's like no one is scared of viruses everyone just hates them because there's not much you can do about them except the things you are already. If a bad thing is happening and you have no agency it won't cause fear, only despair. It's torture not suspense.

**Comment by Formerxboxgamer** [comment]
so for 6 TC I can stun a sec officer, and prevent him from calling for backup because the rest of his department is now stunned and deaf, so they cant hear him dying. This reminds me of the old para pen item.

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -18,13 +18,14 @@
 
 /mob/living/carbon/get_ear_protection()
 	. = ..()
-	if(HAS_TRAIT(src, TRAIT_DEAF))
-		return INFINITY //For all my homies that can not hear in the world
-	var/obj/item/organ/internal/ears/E = get_organ_slot(ORGAN_SLOT_EARS)
-	if(!E)
+	var/obj/item/organ/internal/ears/ear = get_organ_slot(ORGAN_SLOT_EARS)
+	if(!ear)
```
Comment (san7890) [inline code/modules/mob/living/carbon/carbon_defense.dm:22]: isnull

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -473,33 +474,31 @@
...
+			to_chat(src, span_warning("Your ears start to ring badly!"))
+			if(prob(ears.damage - 5))
+				to_chat(src, span_userdanger("You can't hear anything!"))
+				// Makes you deaf, enough that you need a proper source of healing, it won't self heal
+				// you need earmuffs, inacusiate, or replacement
+				ears.set_organ_damage(ears.maxHealth)
+		else if(ears.damage >= 5)
+			to_chat(src, span_warning("Your ears start to ring!"))
+		SEND_SOUND(src, sound('sound/weapons/flash_ring.ogg',0,1,0,250))
+	return effect_amount //how soundbanged we are
```
Comment (san7890) [inline code/modules/mob/living/carbon/carbon_defense.dm:501]: is it possible to have some early returns here? the indentation gets really bad

**Review by san7890 — CHANGES_REQUESTED** [review]
if i had it my way (which i don't because i'm abdicating control over this) this would be a clown ops only item but i still think it's a cool idea so i won't condemn it to that fate

**Comment by Dorsisdwarf** [comment]
> The concerns about this are way overblown. An 8-second stun, much less a 6-second stun, is enough to barely kill ONE person. The rest will simply be prevented from intervening while you do it. So it’s one very tentative kill confirmed for 6 TC, same thing a sleepypen could do basically.

An 8 second stun timer is kinda enough time to crit fully-armored person by wailing on their armored chest using nothing but a standard toolbox. A sleepypen also isn't instant and doesn't stun and deafen the rest of the department preventing them from calling to help

**Comment by SomeoneYouProbablyKnow** [comment]
Yes but you CAN do something. You are limited in what you can do but you till have agency. Even basic stunbatons let you do SOMETHING like call for help. This literally prevents you from doing anything, talking, moving, fighting, running all are out of the question. 
Maybe there's some obscure thiing you can do like open your pda tab and send a message while stunned and deafened which people will find out about and start using but the point is there's nothing in the moment you can do about. That's not scary not dangerous not interesting just plain annoying

**Comment by MrMelbert** [comment]
> The concerns about this are way overblown. An 8-second stun, much less a 6-second stun, is enough to barely kill ONE person. The rest will simply be prevented from intervening while you do it. So it’s one very tentative kill confirmed for 6 TC, same thing a sleepypen could do basically.

Sinful you play this game, if I have an esword 8 seconds is more than enough to crit AT LEAST 2 people. Even ignoring the even stronger weapons available like the revolver or desword. Team up with 1 other traitor and that's that.

**Comment by Jacquerel** [comment]
If it helps I could make it leave your headset on so you can yell deafly into the common radio channel.
I'll be honest that's an element I mostly added because I thought it was funny.

**Comment by SomeoneYouProbablyKnow** [comment]
It is quite funny, can always make it so it turns on the mic when it falls off. Although i don't know how to explain that diogetically.
Although headsets falling off does open up space for flashbang cheese, get a box, stun the whole brig and keep stunning them while you kill them all. It IS funny so i'd say keep it.

**Comment by Deadgebert** [comment]
> The concerns about this are way overblown. An 8-second stun, much less a 6-second stun, is enough to barely kill ONE person. The rest will simply be prevented from intervening while you do it. So it’s one very tentative kill confirmed for 6 TC, same thing a sleepypen could do basically.
> 
> I think 6 TC is a bit steep. For a 6 second stun on a department, which to be honest won’t really let you do much but serve as an annoyance and maybe as a funky way of ambushing someone. I think you had it right the first time with 4 TC. No nukeops is a good call tho.

3 or 6 seconds might not be enough to outright kill, but it's more than enough time to guarantee a stam crit, which is basically a guaranteed kill.
The biggest issue is that there is no real counterplay, you just die. There is nothing to be learned from it other than carry a bounced radio from now on.

**Comment by chemistrymain3** [comment]
the other downside is that i see it being used as 'annoying grief item' like crab-17 is

less impactful, perhaps, but plenty of people are more than willing to spend a few tc on the crab-17 just to be annoying: i could see a bunch of people just buying it and blowing people's ears out for cheap laughs any time there's something important going on  like a blob/just because

**Comment by san7890** [comment]
yeah that's a problem with a lot of cool funny items is that people will run the gimmick into the mud with zero recourse and force people to balance against their overutilization

**Comment by wesoda25** [comment]
This beats detomax, but suffers from the same issues imo. That is, the ability to remotely weaponize essential equipment with no required effort. What if, in order to work, this required telecomms access in some fashion? Not to activate, but to insert some sort of "malware" into the system or something, which then enables you to press your button and temporarily incapacitate a department?

**Comment by ExcessiveUseOfCobblestone** [comment]
How do you avoid it? Like is the biggest sin of the detomatrix REALLY that it blows up your ID/PDA and not the fact that its just a 1 click wonder for someone to insta-win a fight?

Be a (wo)man, remove the item 😈

**Comment by ATH1909** [comment]
> How do you avoid it? Like is the biggest sin of the detomatrix REALLY that it blows up your ID/PDA and not the fact that its just a 1 click wonder for someone to insta-win a fight with no recourse?

doesn't the reworked detomatix require you to reply to a specific (custom) message? didn't that make it unviable for mid-combat picks?

**Comment by SinfulBliss** [comment]
> > The concerns about this are way overblown. An 8-second stun, much less a 6-second stun, is enough to barely kill ONE person. The rest will simply be prevented from intervening while you do it. So it’s one very tentative kill confirmed for 6 TC, same thing a sleepypen could do basically.
> 
> Sinful you play this game, if I have an esword 8 seconds is more than enough to crit AT LEAST 2 people. Even ignoring the even stronger weapons available like the revolver or desword. Team up with 1 other traitor and that's that.

You could also buy a minibomb and chuck it at two people to crit them for like 6 TC (also with no counterplay)

**Comment by Loafin34** [comment]
@SinfulBliss You can toss the bomb back. There's also enough time before the bomb goes off that you can yell "JOHN GREYTIDE TO-GACK"

**Comment by SomeoneYouProbablyKnow** [comment]
It's not about if you CAN do it. It's about the oh oh moment. Someone pulls out a desword and OH OH now you are dead. Someone throws a minibomb and OH OH you are dead those few seconds make it feel fair even if realistically you couldn't have done anything. That's why people hate cult stun, because they pull it out and you are done for, if they do it right you don't get a chance to react or process it.
If someone comes around the corner with an instakill rifle and blasts me that's funny, if an admin just walks up to me and presses the smite button it's kinda infuriating. It's those few seconds, they make the difference. Who cares about balance, this is about player experience.

Maybe this earblast device should have some delay built into it? Not for people to avoid it but to process what's happening. Enough time to yell "oh shit", you know? Idk how it would make sense? Maybe it plays a weird hypno esuqe sound and even if you throw out your radio it still gets you in a moment? Maybe it's a brown note situation, vomiting is let's be honest as annoying as a stun but funnier

**Comment by ATH1909** [comment]
> Maybe this earblast device should have some delay built into it? Not for people to avoid it but to process what's happening. Enough time to yell "oh shit", you know? Idk how it would make sense? Maybe it plays a weird hypno esuqe sound and even if you throw out your radio it still gets you in a moment? Maybe it's a brown note situation, vomiting is let's be honest as annoying as a stun but funnier

adding a delay between pressing the button on the device and it triggering (but not necessarily a delay between the sound and the stun) could also work, so that if you pop it in combat, the other guy can SEE you pull it out and knows that he should take his headset off bc you probably pressed the button

... although that does kind of break down if you just have a buddy hold it and watch you on cams, I guess

**Comment by Jacquerel** [comment]
Doesn't seem to be any interest in merging this and people keep bumping it while I am waiting for stalebot to close it.
I'll maybe PR detomatix removal separately and try cooking a different item, if we even need a replacement.

**Pull request closed without merging**